src/frontends/tensorflow_common/src/op/gather.cpp

```diff
@@ -28,7 +33,22 @@ OutputVector translate_basic_gather_op(const NodeContext& node, const ov::Output
 OutputVector translate_gather_op(const NodeContext& node) {
     // Gather has two inputs: data and indices
     // axis by which data is sliced is always equal to 0, batch_dims is always equal to 0
-    default_op_checks(node, 2, {"Gather"});
+    default_op_checks(node, 2, {"Gather"}, true);
+    auto params = node.get_input(0);
+    auto complex_type_mark = as_type_ptr<ComplexTypeMark>(params.get_node_shared_ptr());
+
+    if (complex_type_mark) {
+        params = complex_type_mark->input_value(0);
+        // Since complex type outputs would have an extra axis for Real & Image, we route the indicies to
+        // be picked from the 2nd dimension instead
+        auto axis = make_shared<v0::Constant>(element::i64, Shape{}, 1);
+        auto indices = node.get_input(1);
+        auto gather = make_shared<v8::Gather>(params, indices, axis, 0);
+        set_node_name(node.get_name(), gather);
+        auto complex_reshape = make_shared<ComplexTypeMark>(gather, complex_type_mark->get_complex_part_type());
+        return {complex_reshape->output(0)};
```

Comment:
implementation is incorrect. Please think how complexity affects `Gather` operation if auxiliary dimension is the last one

Reply:
Hi @rkazants,
I wanted to clear a doubt regarding my understanding of what we mean by "represent output complex type tensor as a floating-point type tensor with auxiliary dimension that concatenates real and imaginary parts of complex tensor."

For example - if we say our complex tensor is of something like [[1+2j 3+4j],[5+6j, 7+8j]] which is of shape (3,2) - The representative floating-point type tensor would be of shape (3,2,2) or (3,4)? Would be great if you could help me understand 

Regards,
Monal Singh

Reply:
Sure, it will be of shape `(3,2,2)`